### ISSUE TITLE
Returning real time waveform

### ISSUE THREAD

**Issue by m15kh** [juanmc2005/diart#250]
Hi,

Regarding this issue: [link](https://github.com/juanmc2005/diart/issues/119), it returns annotations in real time.

How can I achieve the same for sound? I want to save the exact waveform in WAV format in real time.

**Comment by yeahphil** [comment]
That audio is what you're feeding in to the model -- you'll want to tee it before, not try and get audio back out alongside predictions.

**Comment by m15kh** [comment]
@yeahphil 
Please see the image below:
![image](https://github.com/user-attachments/assets/6f778a6e-85a6-45b7-b4f1-5e5b24ef6e05)
In the image above, my custom model detects something as I intended. However, I would like the model to save the exact audio corresponding to the predicted label (highlighted in blue) when it makes a prediction.

How can I achieve this?

**Comment by juanmc2005** [comment]
Is this a duplicate of #246 ?
I just posted an answer there that should help you get the bits of audio alongside the predictions.
Let me know how it works.

**Comment by m15kh** [comment]
@juanmc2005  
Yes it's duplicate question becuase i thought i haden't explained well in issue [246](https://github.com/juanmc2005/diart/issues/246)

first of all thanks for your answering

i did this i now i can save a segment audio that model predicted

```
count = 0
def use_real_time_prediction(results):
    global count 
    prediction, waveform = results  
    
    if prediction:
        
        print(waveform.data)
        
        #NOTE for save audio
        filename = f'output/waveform{count}.wav'  # Using f-string for dynamic filename
        sf.write(filename, waveform.data, samplerate=16000)  # Replace 16000 with the actual sample rate if different
        print(f"Waveform saved to {filename}")
        count += 1  # Increment count



config = PipelineConfig
path_model  = 'checkpoint.ckpt'
config = VoiceActivityDetectionConfig(segmentation=m.SegmentationModel.from_pyannote(path_model))
pipeline = VoiceActivityDetection(config=config)
mic = MicrophoneAudioSource()
inference = StreamingInference(pipeline, mic, do_plot=False, do_profile=True) 

inference.attach_hooks(use_real_time_prediction)

total_prediction = inference()
```

**Comment by m15kh** [comment]
@juanmc2005 
i also added this feature and sent  request for pull requests
